lib/pbench/server/__init__.py

```diff
@@ -157,29 +158,63 @@ def _ref_datetime(self):
 
     @property
     def rest_uri(self):
-        return self._get_conf("pbench-server", "rest_uri")
+        return self.get_conf("pbench-server", "rest_uri")
 
     @property
     def max_retention_period(self) -> int:
+        """Produce an integer representing the maximum number of days the server
+        allows a dataset to be retained.
+
+        Returns:
+            integer number of days representing retention period
         """
-        Produce a timedelta representing the number of days the server allows
-        a dataset to be retained.
+        retention_days = self.conf.getint(
+            "pbench-server",
+            "maximum-dataset-retention-days",
+            fallback=self.MAXIMUM_RETENTION_DAYS,
+        )
+        return retention_days
+
+    @property
+    def default_retention_period(self) -> int:
+        """Produce an integer representing the default number of days the server
+        allows a dataset to be retained.
 
-        Returns
-            delta time representing retention period
+        Returns:
+            integer number of days representing retention period
         """
-        try:
-            retention_days = int(
-                self._get_conf("pbench-server", "maximum-dataset-retention-days")
-            )
-        except (NoOptionError, NoSectionError):
-            retention_days = self.MAXIMUM_RETENTION_DAYS
+        retention_days = self.conf.getint(
+            "pbench-server",
+            "default-dataset-retention-days",
+            fallback=self.DEFAULT_RETENTION_DAYS,
+        )
         return retention_days
 
-    def _get_conf(self, section, option):
+    @property
+    def rest_max_content_length(self) -> int:
+        """Produce an integer representing the maximum content length accepted
+        by the REST APIs.
+
+        Returns:
+            integer number of bytes
         """
-        _get_conf - get the option from the section, raising
-        BadConfig Error if it is empty, return the option value as a string.
+        rest_max_content_length = filesize_bytes(
+            self.conf.get("pbench-server", "rest_max_content_length", fallback="1 gb")
+        )
+        return rest_max_content_length
+
+    def get_conf(self, section: str, option: str) -> str:
+        """Get the option from the section.
```

Comment:
Some initial thoughts:

 1. We don't to affect the agent code by a behavior change to `PbenchConfig.get()`
 2. Further refactoring of the behaviors would cause this PR to expand further so if this is a big concern somebody should file a PR to follow up with these kinds of changes
 3. I am in favor of being too magical unless it solves a problem in the code; this change is an improvement because we are not catching empty configuration values as well

Reply:
I think there's a word missing between "don't" and "to" in your first thought, above.  And, I suspect that the word "not" is missing between "am" and "in" in your third, but I can't be sure.  🙂  I'm wondering if there might be other words missing as well....  😉

I concur that we certainly don't want to break the Agent code with Server changes.  However, if we improve these foundations, the Agent will potentially benefit.

As for your other two thoughts, we'll have to see what this PR looks like once it's been rebased on #3179.